src/mon/PGMap.cc

```diff
@@ -3244,13 +3244,29 @@ void PGMap::get_health_checks(
         ostringstream ss;
         ss << detail.size() << " pgs not scrubbed for " << age;
         auto& d = checks->add("PG_NOT_SCRUBBED", HEALTH_WARN, ss.str());
-        d.detail.swap(detail);
+
+        std::copy_n(detail.begin(),
+                    std::min(detail.size(), (long unsigned int)max),
+                    std::back_inserter(d.detail));
```

Comment:
Can you do something like https://github.com/david-z/ceph/blob/a3df00e364afce616f697039b55bf4bdca3f31fd/src/mon/PGMap.cc#L3453 instead?  Otherwise we generate a potentially huge list of 100k+ pgs only to throw it all away.